src/tzdb.rs

```diff
@@ -27,28 +27,114 @@
 // where
 // offset diff = is_dst { dst_off - std_off } else { std_off - dst_off }, i.e. to_offset - from_offset
 
-use std::path::{Path, PathBuf};
+use std::path::Path;
+#[cfg(not(target_os = "windows"))]
+use std::path::PathBuf;
 
 use alloc::collections::BTreeMap;
 use alloc::string::{String, ToString};
 use alloc::{vec, vec::Vec};
 use core::cell::RefCell;
 
+use combine::Parser;
+
 use tzif::{
     self,
     data::{
-        posix::PosixTzString,
+        posix::{DstTransitionInfo, PosixTzString, TransitionDay, ZoneVariantInfo},
         time::Seconds,
         tzif::{DataBlock, LocalTimeTypeRecord, TzifData, TzifHeader},
     },
 };
 
-use crate::{components::tz::TzProvider, iso::IsoDateTime, TemporalError, TemporalResult};
+use crate::{components::tz::TzProvider, iso::IsoDateTime, utils, TemporalError, TemporalResult};
 
+#[cfg(not(target_os = "windows"))]
 const ZONEINFO_DIR: &str = "/usr/share/zoneinfo/";
 
-pub type TransitionInfo = [Option<LocalTimeTypeRecord>; 2];
+/// `LocalTimeRecord` represents an local time offset record.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct LocalTimeRecord {
+    /// Whether the local time record is a Daylight Savings Time.
+    pub is_dst: bool,
+    /// The time zone offset in seconds.
+    pub offset: i64,
+}
+
+impl LocalTimeRecord {
+    fn from_dst(info: &ZoneVariantInfo) -> Self {
+        Self {
+            is_dst: true,
+            offset: -info.offset.0,
+        }
+    }
+
+    fn from_std(info: &ZoneVariantInfo) -> Self {
+        Self {
+            is_dst: false,
+            offset: -info.offset.0,
+        }
+    }
+}
```

Comment:
Maybe `from_standard_time` and `from_daylight_time` instead. The short names are pretty confusing at first glance, especially `from_std` which is commonly used to convert an `std` type into a library type.

Reply:
Yeah, that's probably a fair point. I can switch it over.